noaa.go

```diff
@@ -33,90 +33,147 @@ type PointsResponse struct {
+	Min            float64 `json:"minValue"`
+	UnitCode       string  `json:"unitCode"`
+	QualityControl string  `json:"qualityControl"`
+}
+
+// Period holds the JSON values for a period within a forecast response's periods.
+type ForecastResponsePeriod struct {
+	ID              int32   `json:"number"`
+	Name            string  `json:"name"`
+	StartTime       string  `json:"startTime"`
+	EndTime         string  `json:"endTime"`
+	IsDaytime       bool    `json:"isDaytime"`
+	Temperature     float64 `json:"temperature"`
+	TemperatureUnit string  `json:"temperatureUnit"`
+	WindSpeed       string  `json:"windSpeed"`
+	WindDirection   string  `json:"windDirection"`
+	Icon            string  `json:"icon"`
+	Summary         string  `json:"shortForecast"`
+	Details         string  `json:"detailedForecast"`
+}
+
+type ForecastResponsePeriodHourly struct {
+	ID               int32   `json:"number"`
+	Name             string  `json:"name"`
+	StartTime        string  `json:"startTime"`
+	EndTime          string  `json:"endTime"`
+	IsDaytime        bool    `json:"isDaytime"`
+	TemperatureTrend *string `json:"temperatureTrend"`
+	WindDirection    string  `json:"windDirection"`
+	Summary          string  `json:"shortForecast"`
+	Icon             string  `json:"icon,omitempty"`
+	Details          string  `json:"detailedForecast"`
+}
+
 // ForecastResponse holds the JSON values from /gridpoints/<cwa>/<x,y>/forecast"
 type ForecastResponse struct {
 	// capture data from the forecast
-	Updated   string `json:"updated"`
-	Units     string `json:"units"`
-	Elevation struct {
-		Value float64 `json:"value"`
-		Units string  `json:"unitCode"`
-	} `json:"elevation"`
-	Periods []struct {
-		ID              int32   `json:"number"`
-		Name            string  `json:"name"`
-		StartTime       string  `json:"startTime"`
-		EndTime         string  `json:"endTime"`
-		IsDaytime       bool    `json:"isDaytime"`
-		Temperature     float64 `json:"temperature"`
-		TemperatureUnit string  `json:"temperatureUnit"`
-		WindSpeed       string  `json:"windSpeed"`
-		WindDirection   string  `json:"windDirection"`
-		Summary         string  `json:"shortForecast"`
-		Details         string  `json:"detailedForecast"`
-	} `json:"periods"`
-	Point *PointsResponse
+	Updated   string                    `json:"updated"`
+	Units     string                    `json:"units"`
+	Elevation *ForecastElevation        `json:"elevation"`
+	Periods   []*ForecastResponsePeriod `json:"periods"`
+	Point     *PointsResponse
+}
+
+// WeatherValueItem holds the JSON values for a weather.values[x].value.
+type WeatherValueItem struct {
+	Coverage  string `json:"coverage"`
+	Weather   string `json:"weather"`
+	Intensity string `json:"intensity"`
+}
+
+// WeatherValue holds the JSON value for a weather.values[x] value.
+type WeatherValue struct {
+	ValidTime string              `json:"validTime"` // ISO 8601 time interval, e.g. 2019-07-04T18:00:00+00:00/PT3H
+	Value     []*WeatherValueItem `json:"value"`
```

Comment:
Ditto, make this a slice of structs instead of pointers to structs.

Reply:
Ack.